src/transformStateWithClones.js

```diff
@@ -7,7 +7,89 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const actionTypes = [];
+  let REPEAT = false;
+
+  for (const action of actions) {
+    if (!actionTypes.includes(action.type)) {
+      actionTypes.push(action.type);
+    }
+  }
+
+  if (actionTypes.length === 1) {
+    REPEAT = true;
+  }
+
+  const currentState = { ...state };
+  const stateHistory = [];
+  const extraData = {};
+  const keysToRemove = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    let copyState = {};
```

Comment:
The variable `copyState` is being redeclared inside the loop for each iteration, which is unnecessary. You can declare it once outside the loop and then just reassign its value inside the loop when needed.